bayesian/utility/db_gateway.py

```diff
@@ -72,6 +80,27 @@ def get_ca_data_from_graph(cls, ecosystem, package, version, vendor):
         logger.info('Gremlin request took %f seconds', (datetime.now() - start).total_seconds())
         return query_result.json()
 
+    @classmethod
+    def get_batch_ca_data(cls, ecosystem: str, packages) -> dict:
+        """Component Analyses Batch Call."""
+        logger.debug('Executing get_batch_ca_data')
+        payload = {
+            'gremlin': cls.ca_batch_query,
+            'bindings': {
+                'ecosystem': ecosystem,
+                'packages': packages
+            }
+        }
+        try:
+            started_at = time.time()
+            response = post(url=gremlin_url, data=json.dumps(payload))
+            response.raise_for_status()
+            elapsed_time = time.time() - started_at
+            logger.info("It took %s to fetch results from Gremlin." % elapsed_time)
+            return response.json()
+        except Exception as e:
+            logger.exception(e)
```

Comment:
Dont you want to raise an exception and prevent further processing if the graph query fails?

Reply:
Encapsulated Parent caller in `api_v2.py` in try: catch. The exception here was kind of redundant. Removed.